apiserver/params/crossmodel.go

```diff
@@ -264,9 +264,17 @@ type RemoteRelationResults struct {
 // RemoteApplication describes the current state of an application involved in a cross-
 // model relation, from the perspective of the local environment.
 type RemoteApplication struct {
-	Name   string `json:"name"`
-	Life   Life   `json:"life"`
+	// Name is the name of the application.
+	Name string `json:"name"`
+
+	// Life is the current lifecycle state of the application.
+	Life Life `json:"life"`
+
+	// Status is the current status of the application.
 	Status string `json:"status"`
+
+	// ModelUUID is the UUId of the model hosting the application.
```

Comment:
I looked at how the JAAS code works and you make an API connection to the model facade on which you want to operate - the api code redirects as needed to the right controller. You still need the model uuid to do that. You get an the api connection info, set the model uuid, and open the api connection - the login process will redirect if the target controller is not the one hosting the model. From there, the worker just makes an api call to the facade which under the covers is connected to the correct target model.